src/renderer/views/Playlist/Playlist.js

```diff
@@ -583,16 +583,49 @@ export default defineComponent({
       }
     },
 
-    removeVideoFromPlaylist: function (videoId, playlistItemId) {
+    removeVideoFromPlaylist: function (videoId) {
       try {
-        this.removeVideo({
-          _id: this.playlistId,
-          videoId: videoId,
-          playlistItemId: playlistItemId,
+        // Initialize deletedVideoIds array if not already
+        if (!this.deletedVideoIds) {
+          this.deletedVideoIds = []
+        }
+        const playlistItems = [].concat(this.playlistItems)
+        const tempPlaylistItems = [].concat(this.playlistItems)
+        let isUndoClicked = false
+
+        const videoIndex = this.playlistItems.findIndex((video) => {
+          return video.videoId === videoId
         })
-        // Update playlist's `lastUpdatedAt`
-        this.updatePlaylist({ _id: this.playlistId })
-        showToast(this.$t('User Playlists.SinglePlaylistView.Toast.Video has been removed'))
+
+        if (videoIndex !== -1) {
+          // Add the video to the deletedVideoIds list
+          this.deletedVideoIds.push(this.playlistItems[videoIndex].videoId)
+          playlistItems.splice(videoIndex, 1)
+          this.playlistItems = playlistItems
+
+          if (!this.lastToastTime || Date.now() - this.lastToastTime > 5000) {
```

Comment:
I wonder if `Date.now() - this.lastToastTime > 5000` is needed...
Should the code just clear `lastToastTime` after `5000`?
```suggestion
          // Only show toast when no existing toast shown
          if (!this.lastToastTime || Date.now() - this.lastToastTime > 5000) {
```